### PR TITLE
Stdout exporter for span and metrics

### PR DESCRIPTION
**Pull request by zeitlinger**
Follow up to https://github.com/open-telemetry/opentelemetry-java/pull/6675

### CODE REVIEW

**Comment by zeitlinger** [comment]
@jack-berg here's the PR that brings metrics and spans to the same level as logs were before 

memory mode will be the final PR

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterBuilderUtil.java [hunk 1]
```diff
@@ -96,5 +98,95 @@ public static void configureHistogramDefaultAggregation(
...
   }
 
+  /**
+   * Invoke the {@code aggregationTemporalitySelectorConsumer} with the configured {@link
+   * AggregationTemporality}.
+   */
+  public static void configureOtlpAggregationTemporality(
+      ConfigProperties config,
+      Consumer<AggregationTemporalitySelector> aggregationTemporalitySelectorConsumer) {
+    String temporalityStr = config.getString("otel.exporter.otlp.metrics.temporality.preference");
```
Comment (jack-berg) [inline exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterBuilderUtil.java:108]: These properties are defined [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/otlp.md) in the context of the OTLP network protocol exporters. Its not clear to me based on today's spec that they would also be applicable to the [proposed `otlp/stdout` exporter](https://github.com/open-telemetry/opentelemetry-specification/pull/4183/files#diff-6e0e0a1cfd634265edad8e3c862f059d3a6f3f61d25fe21df714cfc96764a0b5). For the purposes of the java implementation, let's punt on this until the spec is clear.

If you feel strongly about this, consider extending your spec PR to make it clear that these properties are in fact applicable to `otlp/stdout`. 

This applies to `otel.exporter.otlp.metrics.temporality.preference` and `otel.exporter.otlp.metrics.default.histogram.aggregation`.

File: exporters/logging-otlp/src/main/java/io/opentelemetry/exporter/logging/otlp/internal/metrics/OtlpStdoutMetricExporter.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Exporter for sending OTLP log records to stdout.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public class OtlpStdoutMetricExporter implements MetricExporter {
```
Comment (jack-berg) [inline exporters/logging-otlp/src/main/java/io/opentelemetry/exporter/logging/otlp/internal/metrics/OtlpStdoutMetricExporter.java:31]:
```suggestion
public final class OtlpStdoutMetricExporter implements MetricExporter {
```
Applies to the log, trace equivalents as well.

File: exporters/logging-otlp/src/main/java/io/opentelemetry/exporter/logging/otlp/internal/metrics/OtlpStdoutMetricExporter.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import io.opentelemetry.sdk.metrics.export.DefaultAggregationSelector;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.util.Collection;
+import java.util.StringJoiner;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Exporter for sending OTLP log records to stdout.
```
Comment (jack-berg) [inline exporters/logging-otlp/src/main/java/io/opentelemetry/exporter/logging/otlp/internal/metrics/OtlpStdoutMetricExporter.java:26]:
```suggestion
 * Exporter for sending OTLP metrics records to stdout.
```
Copy paste error. Applies to trace equivalent as well.

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterBuilderUtil.java (hunk 1 above)
Comment (zeitlinger) [reply exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterBuilderUtil.java:108]: The spec PR has this to make it clear that these properties are MUST:

https://github.com/open-telemetry/opentelemetry-specification/pull/4183/files#diff-6e0e0a1cfd634265edad8e3c862f059d3a6f3f61d25fe21df714cfc96764a0b5R27-R29

**Review by jack-berg — APPROVED** [review]

**Pull request merged**